packages/frontend/src/components/ChartConfigPanel/Grid/index.tsx

```diff
@@ -14,42 +13,87 @@ export const defaultGrid: EchartsGrid = {
     bottom: '30', // pixels from bottom (makes room for x-axis)
 };
 
+const positions = ['top', 'bottom', 'left', 'right'] as const;
+
+type Units = '%' | 'px';
+
 const GridPanel: FC = () => {
     const {
         cartesianConfig: { dirtyEchartsConfig, setGrid },
     } = useVisualizationContext();
-    const methods = useForm<EchartsGrid>({
-        mode: 'onBlur',
-        defaultValues: { ...defaultGrid, ...dirtyEchartsConfig?.grid },
+
+    const [config, setConfig] = useState<EchartsGrid>({
+        ...defaultGrid,
+        ...dirtyEchartsConfig?.grid,
     });
 
+    const handleUpdate = (
+        position: typeof positions[number],
+        value: string,
+        currentUnit: Units,
+    ) => {
+        setConfig((prevState) => {
+            const newState = {
+                ...prevState,
+                [position]: `${value}${currentUnit}`,
+            };
+            setGrid(newState);
+            return newState;
+        });
+    };
+
+    const handleUpdateUnit = (key: typeof positions[number], unit: Units) => {
+        const value = config[key];
+        const pureValue = value?.replace(unit === 'px' ? '%' : 'px', '') ?? '';
+
+        if (!pureValue) return;
+
+        handleUpdate(key, pureValue, unit);
+    };
+
     return (
-        <Form
-            name="grid"
-            methods={methods}
-            onSubmit={() => undefined}
-            onBlur={methods.handleSubmit(setGrid)}
-        >
-            <SectionTitle>Margins</SectionTitle>
-            <SectionRow>
-                <Input name="top" label="Top" placeholder={defaultGrid.top} />
-                <Input
-                    name="bottom"
-                    label="Bottom"
-                    placeholder={defaultGrid.bottom}
-                />
-                <Input
-                    name="left"
-                    label="Left"
-                    placeholder={defaultGrid.left}
-                />
-                <Input
-                    name="right"
-                    label="Right"
-                    placeholder={defaultGrid.right}
-                />
-            </SectionRow>
-        </Form>
+        <SectionRow>
+            {positions.map((position) => {
+                const value = config[position];
+                const currentUnit = value?.includes('%') ? '%' : 'px';
+                const nextUnit = currentUnit === '%' ? 'px' : '%';
+                const valueWithoutUnit = value?.replace(/%|px/g, '');
+
+                return (
+                    <FormGroup
+                        key={position}
+                        label={startCase(position)}
+                        labelFor={`${position}-input`}
+                    >
+                        <InputGroup
+                            id={`${position}-input`}
+                            name={position}
+                            placeholder={defaultGrid[position]}
```

Comment:
We need to parse the placeholders values as well
![Screenshot 2022-11-14 at 18 30 33](https://user-images.githubusercontent.com/9117144/201738638-99fbeeca-7b4a-466e-bbbe-88d639be0da5.png)